### ISSUE TITLE
Typescript migrations doesn't work

### ISSUE THREAD

**Issue by andrejleitner** [sequelize/cli#1053]
## What you are doing?
I have migrations written in typescript. These compiled worked fine until CLI of version `6.1.0`. They don't after updating to version `6.4.1`.

```bash
npx sequelize db:migrate --env development
```
## What is actually happening?
```bash
ERROR: Cannot use import statement outside a module
```

__Dialect:__ mysql
__Database version:__ 5.7
__Sequelize CLI version:__ 6.4.1
__Sequelize version:__ 6.15.1

**Comment by ephys** [comment]
What does your TypeScript transpilation setup look like? Do you transpile your TS migrations to JS before running them?

It may be caused by this PR https://github.com/sequelize/cli/pull/915/, which loads TypeScript files but, unless ts-node is used, will treat them as plain JavaScript.
IMO that PR should be rolled back and we should instead provide a way to configure which file extensions will be loaded & document how to use with ts-node.

**Comment by andrejleitner** [comment]
> What does your TypeScript transpilation setup look like? Do you transpile your TS migrations to JS before running them?

Yep, I do (as it was necessary before the typescript was supported).

**Comment by ephys** [comment]
It's the .ts files that should be removed from the build output. At least until we have a solution

**Comment by andrejleitner** [comment]
> IMO that PR should be rolled back and we should instead provide a way to configure which file extensions will be loaded & document how to use with ts-node.

Hey @ephys, any progress in here?☝🏻

**Comment by andrejleitner** [comment]
Summoning also @WikiRik, @sdepold, @sushantdhiman as you are probably the last active contributor here and "_feat: support migrations files with ts file extension_" is not actually true.

**Comment by ephys** [comment]
> Hey @ephys, any progress in here?☝🏻

Unfortunately my work on the CLI is limited to issue and PR review as the time I can donate to this project is limited and my focus is on the [main library](github.com/sequelize/sequelize/).

Hopefully someone else is willing to maintain the CLI :/

As for your issue, I think that if you remove the .ts files from your build output, and only keep the transpiled .js files there, it should work.

**Comment by andrejleitner** [comment]
> As for your issue, I think that if you remove the .ts files from your build output, and only keep the transpiled .js files there, it should work.

Yep, but it doesn't actually mean TS migrations are supported. 🙂

**Comment by ristiisa** [comment]
@andrejleitner did you find a solution to this issue?

**Comment by andrejleitner** [comment]
I still transpile `.ts` to `.js` myself.

**Comment by gousta** [comment]
I too am facing the same error and use all kinds of different migrations directory to get around this problem

**Comment by t3chnik** [comment]
Hi! I am too facing same issue 🆙
